pyop2/types/dat.py

```diff
@@ -492,6 +492,20 @@ def norm(self):
         from math import sqrt
         return sqrt(self.inner(self).real)
 
+    def axpy(self, alpha: float, other: 'Dat') -> None:
+        """Compute the operation :math:`y = \\alpha x + y`.
+
+        :arg alpha: a scalar
+        :arg other: the :class:`Dat` to add to this one
+
+        """
+        if isinstance(other._data, np.ndarray):
```

Comment:
I think we should include a check to make sure that the `Dats` are compatible. E.g.:
```py
if self.dataset != other.dataset:
    raise SomeException("NO!!!")
```